stored_file.php

```diff
@@ -0,0 +1,380 @@
+	{
+		if (!$ext) $ext = $this->extension;
+		
+		$hash = md5_file($this->source);
+		$dir1 = substr($hash, 0, 2);
+		$dir2 = substr($hash, 2, 2);
+		$name = substr($hash, 4, 28);
+		
+		$this->filename = build_path($dir1, $dir2,
+			sf('%s.%s', $name, $ext));
+		return $this->filename;
+	}
+	
+	public function read()
+	{
+		return file_get_contents($this->actual_filename());
+	}
+	
+	public function move()
+	{
+		if ($this->__moved) return;
+		$destination = static::resolve_file($this->filename, true);
+		$dir = dirname($destination);
+		if (!is_dir($dir)) mkdir($dir, 0755, true);
+		rename($this->source, $destination);
+		$this->__moved = true;
+
+		// File Tracking Feature Scott U 9/2019
+		// Extracting Path and Hash from file destination
+		$stringParts = explode("/", $destination);
+		$hash = $stringParts[6].$stringParts[7].substr($stringParts[8], 0, (strrpos($stringParts[8], ".")));
+		$path = '/'.$stringParts[1].'/'.$stringParts[2].'/'.$stringParts[3].'/'.$stringParts[4].'/'.$stringParts[5].'/'.$stringParts[6].'/'.$stringParts[7].'/';
+
+		// Setup tracking text file location and temp swap file
+		$trackFile = $path.'tracking.txt';
+		$trackFileTemp = $path.'tracking.temp';
+
+		// Check for existing tracking file
+		if (!file_exists($trackFile)) 
+		{   
+			// tracking text file does NOT exist, create it.
+			// Include initial file uploaded count
+			$content = $hash." = 1";
+
+			// Create tracking text file
+			file_put_contents($trackFile, $content);			
+		}
+		else 
+		{
+			// tracking text file DOES exists
+			// set blank new file content
+			$newFileContent = '';
+
+			// set initial number of matches for hash
+			$found = 0;
+			$tf = fopen($trackFile,'r');
+			
+			// Lock File
+			if (flock($tf, LOCK_EX)) 
+			{
+				// Process each line of tracking 	
+				while ($line = fgets($tf)) 
+				{
+					// check for hash in current line
+				    if (strpos($line, $hash) !== false) 
+				    { 
+				    	// hash found, get existing count
+				    	$existingCount = substr($line, strpos($line, "=") + 1);
+				    	$existingCount = intval(trim($existingCount));
+
+				    	// add upload to count
+				    	$newCount = (int)$existingCount + 1;
+
+				    	// add updated line to new file content
+				        $newFileContent .= $hash." = ".$newCount."\n";
+				    
+				    	// increment number of matches for hash
+				        $found++;   
+				    } 
+				    else 
+				    {
+				    	// add current line to new file content, does not contain hash
+				        $newFileContent .= $line;
+				    }
+				}
+
+				if ($found === 0) 
+				{
+					// hash not in tracking file, add it
+					$newFileContent .= $hash.' = 1';
+				}
+
+			// Save Updates To Temp, Rename Temp to Main
+			file_put_contents($trackFileTemp, $newFileContent);
+			
+			// Unlock trackFile
+			flock($tf, LOCK_UN);
+    		}
+
+			rename($trackFileTemp, $trackFile);
+
+		} // END Else tracking file DOES exist	
+
+		// END File Tracking Feature Scott U 9/2019
```

Comment:
I'm not sure what the point of this comment was. If every developer did this the file would be full of developer names and dates. Comments should be for describing behaviour. Version control (Git) handles names/dates.